Assets/RGLUnityPlugin/Scripts/SceneManager.cs

```diff
@@ -435,5 +465,43 @@ private static IEnumerable<Renderer> GetUniqueRenderersInGameObjects(IEnumerable
             foreach (var smr in smrs) yield return smr;
             foreach (var mr in mrs) yield return mr;
         }
+
+        /// <summary>
+        /// Searches through new rglObjects for ones containing IntensityTexture component.
+        /// If present, check if the found texture was already sent to RGL. 
+        /// If not, send it and write its identifier to sharedTextures dictionary.
+        /// After that assign rglTexture to the proper rglObject.
+        /// </summary>
+        private static void AddTextures(IEnumerable<RGLObject> rglObjects)
+        {
+            foreach (var rglObject in rglObjects)
+            {                
+                var intensityTextureComponent = rglObject.RepresentedGO.GetComponent<IntensityTexture>();
+
+                if( intensityTextureComponent != null)
+                {
+                    if( intensityTextureComponent.texture != null)
+                    {
+                        string textureID = $"r#{intensityTextureComponent.texture.GetInstanceID()}";
```

Comment:
What `r#` stands for?
For me `textureID` could be int.

Reply:
So why RGLMesh Identifier is a string?

Reply:
There are 2 types of meshes: skinned and non-skinned (regular). We wanted to have a distinction.